tests/gtest/avifallocationtest.cc

```diff
@@ -58,22 +58,16 @@ TEST(AllocationTest, MinimumValidDimensions) {
   TestAllocation(1, 1, 8, AVIF_RESULT_OK);
 }
 
-TEST(AllocationTest, MaximumValidDimensions) {
-  // On 32-bit builds, malloc() will fail with fairly low sizes.
-  // Adapt the tests to take that into account.
-  constexpr bool kIsPlatform64b = sizeof(void*) > 4;
-  constexpr uint32_t kMaxAllocatableDimension =
-      kIsPlatform64b ? std::numeric_limits<typeof(avifImage::width)>::max()
-                     : 134217728;  // Up to 1 GB total for YUVA
-
-  // 8 bits
-  TestAllocation(kMaxAllocatableDimension, 1, 8, AVIF_RESULT_OK);
-  TestAllocation(1, kMaxAllocatableDimension, 8, AVIF_RESULT_OK);
-  // 12 bits (impacts the width because avifImage stride is stored as uint32_t)
-  TestAllocation(kMaxAllocatableDimension / 2, 1, 12, AVIF_RESULT_OK);
-  TestAllocation(1, kMaxAllocatableDimension, 12, AVIF_RESULT_OK);
-  // Some high number of bytes that malloc() accepts to allocate.
-  TestAllocation(1024 * 16, 1024 * 8, 12, AVIF_RESULT_OK);  // Up to 1 GB total
+TEST(AllocationTest, MaximumAllocatableValidDimensions) {
+  // Limit the total allocation to 1 GB to prevent any failure no matter the
+  // platform or environment.
```

Comment:
In this test you're only checking that you *can* allocate 1GB, but not that you *can't* allocate more. Shouldn't you test e.g. (1 << 30) / 4 + 1 if (1 << 30) / 4 really is the max valid dimension?
I see other tests in this file testing bad dimensions but they all use numeric_limits::max() instead.

Reply:
It depends on the platform and the environment (build/test farms have lower RAM limits, 32-bit have 1 or 2GB max etc.) so to avoid the pain of having to fine tune this test for each running context, I believe it is simpler to have low passing thresholds and high failing thresholds that work everywhere.